src/core/dbus-manager.c

```diff
@@ -1409,16 +1409,39 @@ static int dump_impl(
 
         _cleanup_free_ char *dump = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        bool check_ratelimit = false;
         int r;
 
         assert(message);
 
-        /* Anyone can call this method */
-
+        /* 'status' access is the bare minimum always needed for this */
         r = mac_selinux_access_check(message, "status", error);
         if (r < 0)
                 return r;
 
+        /* Reload/reexec are also slow but non-destructive, use the same access check for dumps which, given
+         * the large amount of data to fetch, can stall for quite some time. If the caller is not privileged
+         * use rate limiting. */
+        r = mac_selinux_access_check(message, "reload", error);
+        if (r < 0)
+                check_ratelimit = true;
```

Comment:
sorry, but this is wrong. we don't bind behaviour such as ratelimiting to selinux exclusively. selinux is only an *additional* check. never the primary check.

Reply:
I don't follow, there are other 26 such checks in this very same file

Reply:
so, what does that have to do with anything? not following

Reply:
Given we cannot add a new check but have to use an existing one, this is the closest, based on the following logic: it does not change any configuration, it is slow and causes a "stall" in the pid1 main loop. IE, it would be weird to lump this with permission checks that allow modifying units or starting/stopping services. Or is there a better match that I have missed?